Source/SwiftLintFramework/Extensions/File+SwiftLint.swift

```diff
@@ -95,6 +95,34 @@ extension File {
         }
     }
 
+    internal func swiftDeclarationKindsByLine() -> [[SwiftDeclarationKind]]? {
+        if sourcekitdFailed {
+            return nil
+        }
+        var results = [[SwiftDeclarationKind]](repeating: [], count: lines.count + 1)
+        let structureKinds = structure.kinds()
+        var lineIterator = lines.makeIterator()
+        var structureIterator = structureKinds.makeIterator()
+        var maybeLine = lineIterator.next()
+        var maybeStructure = structureIterator.next()
+        while let line = maybeLine, let structure = maybeStructure {
+            if NSLocationInRange(structure.byteRange.location, line.byteRange) ||
+                NSLocationInRange(line.byteRange.location, structure.byteRange) {
```

Comment:
do you really need these two `NSLocationInRange` checks? It seems redundant for me.